src/components/Video/Video.tsx

```diff
@@ -142,20 +142,24 @@ export const Video = ({
   useEffect(() => {
     (async () => {
       if (videoRef.current && hmsVideoTrack) {
-        HMSLogger.d('Video InView', videoTrack, inView);
-        if (hmsVideoTrack.enabled) {
-          if (inView) {
+        HMSLogger.d('Video InView', hmsVideoTrack, inView);
+        if (inView) {
+          if (hmsVideoTrack.enabled) {
+            // attach when in view and enabled
             await hmsActions.attachVideo(hmsVideoTrack.id, videoRef.current);
           } else {
+            // detach when in view but not enabled
             await hmsActions.detachVideo(hmsVideoTrack.id, videoRef.current);
           }
+        } else {
+          // detach when not in view
+          await hmsActions.detachVideo(hmsVideoTrack.id, videoRef.current);
         }
       }
     })();
   }, [
     inView,
     videoRef,
-    videoTrack,
```

Comment:
regarding this, it's possible a client is using this component directly with native track for preview api. just sth to keep in mind if we see any related issue.

Reply:
there is another useEffect that handled videoTrack.